common/src/chain/transaction/signature/tests/sign_and_mutate.rs

```diff
@@ -992,12 +961,42 @@ fn add_input(_rng: &mut impl Rng, tx: &SignedTransactionWithUtxo) -> SignedTrans
     }
 }
 
-fn mutate_input(rng: &mut impl Rng, tx: &SignedTransactionWithUtxo) -> SignedTransactionWithUtxo {
+fn mutate_first_input(
+    rng: &mut impl Rng,
+    tx: &SignedTransactionWithUtxo,
+) -> SignedTransactionWithUtxo {
     let mut updater = MutableTransaction::from(&tx.tx);
-    updater.inputs[0] = TxInput::new(
-        OutPointSourceId::Transaction(Id::<Transaction>::from(H256::random_using(rng))),
-        9999,
-    );
+
+    let mutated_input = match updater.inputs.get(0).unwrap() {
+        TxInput::Utxo(outpoint) => {
+            if rng.gen::<bool>() {
+                TxInput::Utxo(UtxoOutPoint::new(outpoint.tx_id(), rng.gen()))
+            } else {
+                TxInput::Utxo(UtxoOutPoint::new(
+                    OutPointSourceId::Transaction(Id::<Transaction>::from(H256::random_using(rng))),
+                    outpoint.output_index(),
+                ))
+            }
+        }
+        TxInput::Account(outpoint) => {
+            if rng.gen::<bool>() {
+                TxInput::Account(AccountOutPoint::new(
+                    outpoint.nonce(),
+                    AccountSpending::Delegation(
+                        DelegationId::new(H256::random_using(rng)),
+                        Amount::from_atoms(rng.gen()),
+                    ),
+                ))
+            } else {
+                TxInput::Account(AccountOutPoint::new(
+                    AccountNonce::new(rng.gen()),
```

Comment:
Keep in mind that there's a chance the rng provides the same value, and then the test fails.